models/articles.models.js

```diff
@@ -1,10 +1,28 @@
 const db = require("../db/connection.js");
 
 
-exports.fetchArticles = () => {
-    return db
-    .query(`SELECT articles.article_id, articles.author, articles.created_at, articles.title, articles.topic, articles.votes, COUNT(comments.comment_id) AS comment_count
-     FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;`)
+exports.fetchArticles = (sort_by = "created_at", order_by = "DESC", topic) => {
+    if (!['article_id', 'author', 'created_at', 'title', 'topic', 'votes', 'comment_count'].includes(sort_by)) {
+        return Promise.reject({ status: 400, msg: 'Invalid Sort Query' });
+      }
+      if (!['ASC', 'DESC', 'asc', 'desc'].includes(order_by)) {
+        return Promise.reject({ status: 400, msg: 'Invalid Order Query' });
+      }
+      if (!['mitch', 'cats', 'paper', undefined].includes(topic)) {

```

Comment:
What happens when the database gets bigger and there are more topics? This would not get updated and would no longer catch them. 

You need to query the db to check the valid topics dynamically, based on whats in the db.